### PR TITLE
Correct checking for submit in submit_form

### LINKED ISSUE #159: Wrong Submit button detected if form$fields$type is missing
Taking this example form:

```
require(rvest)
ses <- html_session("https://www.handelsregister.de/rp_web/mask.do?Typ=e")
form <- ses %>% html_form() %>% .[[1]]
submit_form(ses, form, submit = 'foo')
```

`submit_form` exits with the error

> Error: Unknown submission name 'foo'.
> Possible values: bundeslandTH, schlagwortOptionen, ort

Looking at the form all three "Possible values" are no submit-buttons:

```
<input checkbox> 'bundeslandTH':
<input radio> 'schlagwortOptionen':
<input text> 'ort':
```

digging into `submit_form` and `submit_request` it turns out that there are `form$fields` that do not have a `form$fields$type`. 

`rvest:::submit_request` reads as follows:

```
is_submit <- function(x) tolower(x$type) %in% c("submit", 
    "image", "button")
  submits <- Filter(is_submit, form$fields)
  if (length(submits) == 0) {
    stop("Could not find possible submission target.", call. = FALSE)
  }
...
```

As the `form$fields$type` is missing `is_submit` returns `logical(0)` which is not at all handled by `Filter`

### PR DESCRIPTION
**Pull request by rentrop**
This fixes #159

### CODE REVIEW

**Comment by rentrop** [comment]
Sorry, but i do have no clue what this error-message means. Dont know how to correct it.

**Comment by hadley** [comment]
Are you still interested in finishing this ok? (Double check since it's been almost 2 years 😱).

**Comment by rentrop** [comment]
Yes, as the test (https://github.com/tidyverse/rvest/pull/161/files#diff-c78a939d1fbf3188452259c9634c97f9) still fails in the current version i think it is still worth to be fixed.

File: tests/testthat/test-forms.R [hunk 1]
```diff
@@ -35,3 +35,24 @@ test_that("parse_fields gets the button", {
...
+                         </select>
+                         <button type="submit" name="clickMe">Click me</button>
+                         </form>
+                         ')
+
+  form <- select %>% html_node("form") %>% html_form()
+  expect_null(form$fields[[1]]$type)
+  expect_equal(form$fields[[2]]$type, "submit")
+
+  req <- rvest:::submit_request(form, "clickMe")
```
Comment (hadley) [inline tests/testthat/test-forms.R]: There's no need to use `rvest:::` inside a test

File: tests/testthat/test-forms.R [hunk 2]
```diff
@@ -35,3 +35,24 @@ test_that("parse_fields gets the button", {
   form <- select %>% html_node("form") %>% html_form()
   expect_equal(form$fields[[1]]$type, "submit")
 })
+
+test_that("submit_request detects submit-buttons correct", {
+  select <- minimal_html("submit test", '
```
Comment (hadley) [inline tests/testthat/test-forms.R]: Can you please indent the html by only 4 spaces?

File: tests/testthat/test-forms.R [hunk 3]
```diff
@@ -35,3 +35,24 @@ test_that("parse_fields gets the button", {
...
+                         <select name="elementWithoutType" size="1">
+                          <option value="10">10</option>
+								          <option value="25">25</option>
+                         </select>
+                         <button type="submit" name="clickMe">Click me</button>
+                         </form>
+                         ')
+
+  form <- select %>% html_node("form") %>% html_form()
+  expect_null(form$fields[[1]]$type)
```
Comment (hadley) [inline tests/testthat/test-forms.R]: I don't think you should be testing your test data here

**Comment by rentrop** [comment]
@hadley I tried to correct it, but i honestly do not understand the Travic CI error:

```
* checking for future file timestamps ...Warning in file(con, "r") :
  cannot open URL 'http://worldclockapi.com/api/json/utc/now': HTTP status was '403 Forbidden'
 WARNING
unable to verify current time
```

How did i cause this bug and how can i correct it? I did not use this `worldclockapi`...

**Comment by hadley** [comment]
It's a bug in R-devel - you don't need to worry about it.

**Comment by hadley** [comment]
Could you please check the box that enables me to edit your pull request? I need to make a few small changes and it'll be easier for me to make directly.

**Comment by rentrop** [comment]
Done

**Pull request merged**